src/nvim/window.c

```diff
@@ -4196,26 +4199,35 @@ int win_new_tabpage(int after, char *filename)
 
     newtp->tp_topframe = topframe;
     last_status(false);
-
     redraw_all_later(UPD_NOT_VALID);
```

Comment:
The `!enter` handling in this function will need a more scrutinous review later.

For example, the window layout is corrupted with `!enter`, as this code has a number of assumptions that we're already in the new tabpage (e.g: `win_alloc_firstwin` overwrites the `topframe`, `firstwin` is changed, etc.).